.github/workflows/build-base.yaml

```
@@ -0,0 +1,48 @@
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+            username: ${{ secrets.DOCKER_USERNAME }}
+            password: ${{ secrets.DOCKER_PUSH_TOKEN }}
+
+      # Build docker image
+      - name: Build docker image
+        uses: docker/build-push-action@v4
```

actionlint (1.7.8)

the runner of "docker/build-push-action@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)